config.go

```diff
@@ -77,6 +77,20 @@ type Config struct {
 	InitialFields map[string]interface{} `json:"initialFields" yaml:"initialFields"`
 }
 
+// ProductionEncoderConfig is an opinionated EncoderConfig for
+// production environments
+var ProductionEncoderConfig = zapcore.EncoderConfig{
```

Comment:
Can you make this and the development config `NewProductionEncoderConfig()/NewDevelopmentEncoderConfig()`?  I fear some user will do:

```
ProductionEncoderConfig.TimeKey = "foo"
...
SomeBug()
```

I don't want this to be mutable. It's very unlikely since in general, zap configs will only be done on setup, but with fx for example, if it does something different/access this in some way, then we could run into a bug.